packages/Toast/ToastWrapper.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { cloneElement } from 'react'
+import toastRHT, { Toast, ToastPosition } from 'react-hot-toast/headless'
+
+import * as S from './styles'
+
+type ToastWrapperProps = {
+  calculateOffset: (
+    toast: Toast,
+    opts?:
+      | {
+          reverseOrder?: boolean | undefined
+          gutter?: number | undefined
+          defaultPosition?: ToastPosition | undefined
+        }
+      | undefined
+  ) => number
+  toast: Toast & CustomToastOptions
+  updateHeight: (toastId: string, height: number) => void
+}
+
+type CustomToastOptions = {
+  onClose?: () => void
+}
+
+export const ToastWrapper: React.FC<ToastWrapperProps> = ({
+  calculateOffset,
+  toast,
+  updateHeight,
+}) => {
+  if (typeof toast.message === 'string') {
+    // eslint-disable-next-line no-console
+    console.warn('You must pass a React component or a HTML element.')
+    return null
+  }
+
+  const offset = calculateOffset(toast, { reverseOrder: false, gutter: 0 })
+
+  const ref = (element: HTMLElement | null) => {
+    if (element && typeof toast.height !== 'number') {
+      const height = element.getBoundingClientRect().height
+      updateHeight(toast.id, height)
+    }
+  }
+
+  const center = toast.position?.includes('center')
+  const top = toast.position?.includes('top')
+  const bottom = toast.position?.includes('bottom')
+
+  const positionStyle = {
```

Comment:
we can move `positionStyle` outside of the component

Reply:
Nice idea!